src/polytopes/segment.jl

```diff
@@ -33,3 +33,20 @@ function (s::Segment)(t)
   a, b = s.vertices
   a + t * (b - a)
 end
+
+function Base.in(p::Point{2}, s::Segment{2})
+  # get vertices and coordinates
+  a, b = s.vertices
+  p1, p2, p3 = coordinates.((a, b, p))
+  # (i)  collinearity
+  arecollinear = (p2[2] - p1[2]) * (p3[1] - p2[1]) - (p3[2] - p2[2]) *
+    (p2[1] - p1[1]) == 0
+  # (ii) compare dot product
+  if arecollinear
+    kap = (b - a) ⋅ (p - a)
+    kab = (b - a) ⋅ (b - a)
```

Comment:
Can you elaborate on the name of these variables? It would be nice to elaborate on what is going on so that future users can easily understand and propose fixes.

Reply:
Given that variables are used only once, I added I description and evaluate the condition directly.